frontend/src/store/modules/auth.js

```diff
@@ -1,42 +1,97 @@
 import axios from 'axios';
 
+import { interceptorRefresh } from '../../main.js';
+
 export default {
   namespaced: true,
   state: {
-    token: localStorage.getItem('token') || ''
+    token: localStorage.getItem('token') || '',
+    error: null
   },
   getters: {
-    isLoggedIn: state => !!state.token
+    isLoggedIn: state => !!state.token,
+    getError(state) {
+      return state.error;
+    }
   },
   mutations: {
-    auth_success(state, token) {
-      state.token = token;
+    raiseError(state, error_msg) {
+      state.error = error_msg;
     }
   },
   actions: {
     login({ commit }, loginData) {
-      return new Promise((resolve, reject) => {
-        axios
-          .post('auth/login/', {
-            username: loginData.username,
-            password: loginData.password
-          })
-          .then(
-            res => {
-              const token = res.data.access_token;
+      delete axios.defaults.headers.common['Authorization'];
+
+      return axios
+        .post('auth/login/', {
+          username: loginData.username,
+          password: loginData.password
+        })
+        .then(res => {
+          const token = res.data.access_token;
+          const refresh_token = res.data.refresh_token;
+
+          localStorage.setItem('token', token);
+          localStorage.setItem('refresh_token', refresh_token);
+          axios.defaults.headers.common['Authorization'] = 'Bearer ' + token;
+          return res.status;
+        })
+        .catch(error => {
+          localStorage.removeItem('token');
+          commit('raiseError', error);
+        });
+    },
+    logout({ dispatch }) {
+      axios.interceptors.response.eject(interceptorRefresh);
+      return axios
+        .delete('auth/logout/')
+        .then(() => {
+          localStorage.removeItem('token');
+        })
+        .catch(() => {
+          localStorage.removeItem('token');
+        })
+        .finally(() => {
+          delete axios.defaults.headers.common['Authorization'];
+          dispatch('revoke');
+        });
+    },
+    revoke() {
+      const refresh_token = localStorage.getItem('refresh_token');
+
+      return axios
+        .delete('auth/refresh-token/', {
+          headers: { Authorization: 'Bearer ' + refresh_token }
+        })
+        .then(() => {
+          localStorage.removeItem('refresh_token');
+        })
+        .catch(() => {
+          localStorage.removeItem('refresh_token');
+        });
+    },
+    refresh() {
+      const refresh_token = localStorage.getItem('refresh_token');
+
+      return axios
```

Comment:
Just general comment, I think this file could use async/await. It's up to you if it's more readable to you

Reply:
Should it looks like that?
```
      async login({ commit }, loginData) {
      delete axios.defaults.headers.common['Authorization'];
      try {
        let res = await axios
        .post('auth/login/', {
          username: loginData.username,
          password: loginData.password
        })
        const token = res.data.access_token;
        const refresh_token = res.data.refresh_token;
        localStorage.setItem('token', token);
        localStorage.setItem('refresh_token', refresh_token);
        axios.defaults.headers.common['Authorization'] = 'Bearer ' + token;
        return res.status;
      } catch(error) {
          localStorage.removeItem('token');
          commit('raiseError', error);
        };
    },
```